### PR TITLE
Return PageRendered in the save layout end point

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 1]
```diff
@@ -308,8 +275,37 @@ public Response renderHTMLOnly(@Context final HttpServletRequest request,
         return res;
     }
 
-    private ImmutableMap<Object, Object> createViewAsMap(final HttpServletRequest request, final User user,
-                                                         final String deviceInode) throws DotDataException {
+    private Map<String, Object> getRenderedPageMap(HttpServletRequest request, HttpServletResponse response, User user,
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java]: These parameters can be declared final

File: dotCMS/src/main/webapp/html/portal/view_portlet_inc.jsp [hunk 2]
```diff
@@ -8,10 +8,14 @@
...
 				request.getParameter(WebKeys.PORTLET_URL_CURRENT_ANGULAR_PORTLET) :
 				portlet.getPortletId();
-		access = APILocator.getLayoutAPI().doesUserHaveAccessToPortlet(accessPortletId,user);
+		access = APILocator.getLayoutAPI().doesUserHaveAccessToPortlet(defaultAccessPortletId,user);
+
+		if (!access && !portlet.getPortletId().equals(defaultAccessPortletId)){
+			access = APILocator.getLayoutAPI().doesUserHaveAccessToPortlet(portlet.getPortletId(),user);
+		}
 	} catch (DotDataException e) {
 		e.printStackTrace();
```
Comment (nollymar) [inline dotCMS/src/main/webapp/html/portal/view_portlet_inc.jsp]: Can this stack trace be logged?

Comment (nollymar) [inline dotCMS/src/main/webapp/html/portal/view_portlet_inc.jsp]: Can this stack trace be logged?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java]: +1

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**